### PR TITLE
📖 Small fixes for getting started overview guide

### LINKED ISSUE #3725: Getting started guide various small typos/fixes
### What broke? What's expected?

The getting started guide can't be followed as is due to a couple of small issues:

https://book.kubebuilder.io/getting-started#create-the-memcached-api-crd
  - group should be 'cache' not example.com
  
https://book.kubebuilder.io/getting-started#define-your-api
  - make manifests it to be used instead of make generate

https://book.kubebuilder.io/getting-started#in-the-context-of-our-example
  - testdata/project-v4-with-deploy-image/internal/controller/memcached_controller.go -> $GOPATH/memcached-operator/internal/controller/memcached_controller.go
 
https://book.kubebuilder.io/getting-started#setting-the-rbac-permissions
  - make manifests it to be used instead of make generate

Will submit a merge request for this.

### Reproducing this issue

Try to run the getting started sample.

### KubeBuilder (CLI) Version

3.13.0

### PROJECT version

_No response_

### Plugin versions

_No response_

### Other versions

_No response_

### Extra Labels

_No response_

### PR DESCRIPTION
**Pull request by nemethloci**
Fixes #3725

Small fixes for getting started guide described in the ticket

### CODE REVIEW

**Comment by nemethloci** [comment]
These are small documentation fixes for: https://github.com/kubernetes-sigs/kubebuilder/issues/3725

**Comment by nemethloci** [comment]
Fixes kubernetes-sigs/kubebuilder#3725

File: docs/book/src/getting-started.md [hunk 1]
```diff
@@ -208,7 +208,7 @@ return ctrl.Result{RequeueAfter: nextRun.Sub(r.Now())}, nil
 
 When a Custom Resource is applied to the cluster, there's a designated controller to manage the Memcached Kind. You can check its reconciliation implemented:
 
-From `testdata/project-v4-with-deploy-image/internal/controller/memcached_controller.go`:
+From `$GOPATH/memcached-operator/internal/controller/memcached_controller.go`:
```
Comment (camilamacedo86) [inline docs/book/src/getting-started.md:211]: We have this example in the dir `testdata/project-v4-with-deploy-image/internal/controller/memcached_controller.go` we might create a link.

File: docs/book/src/getting-started.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ kubebuilder init --domain=example.com
 Next, we'll create a new API responsible for deploying and managing our Memcached solution. In this instance, we will utilize the [Deploy Image Plugin][deploy-image] to get a comprehensive code implementation for our solution.
 
 ```
-kubebuilder create api --group example.com --version v1alpha1 --kind Memcached --image=memcached:1.4.36-alpine --image-container-command="memcached,-m=64,-o,modern,-v" --image-container-port="11211" --run-as-user="1001" --plugins="deploy-image/v1-alpha" --make=false
+kubebuilder create api --group cache --version v1alpha1 --kind Memcached --image=memcached:1.4.36-alpine --image-container-command="memcached,-m=64,-o,modern,-v" --image-container-port="11211" --run-as-user="1001" --plugins="deploy-image/v1-alpha" --make=false
```
Comment (camilamacedo86) [inline docs/book/src/getting-started.md:36]: Are we using cache to generate the example under `testdata/project-v4-with-deploy-image/internal/controller/memcached_controller.go`? Could you please check the shell script that we use? If you want to change it please could you either change the testdata shell and re-run make generate to update the results?

File: docs/book/src/getting-started.md [hunk 3]
```diff
@@ -94,7 +94,7 @@ type MemcachedStatus struct {
 }
 ```
 
-Thus, when we introduce new specifications to this file and execute the `make generate` command, we utilize [controller-gen][controller-gen] to generate the CRD manifest, which is located under the `config/crds` directory.
+Thus, when we introduce new specifications to this file and execute the `make manifests` command, we utilize [controller-gen][controller-gen] to generate the CRD manifest, which is located under the `config/crds` directory.
```
Comment (camilamacedo86) [inline docs/book/src/getting-started.md:97]: +1

**Review by camilamacedo86 — COMMENTED** [review]
Thank you for your contribution 🥇 

For all PRs we need to have a title meaningful and we use emoji to define the type of contribution. I am updating this one and I hope that you do not mind. 

I added a couple of nits as well

**Comment by stmcginnis** [comment]
Sorry, I made the very poor mistake of not taking a look at existing issues and PRs before submitting https://github.com/kubernetes-sigs/kubebuilder/pull/3758. I apologize for duplicating work here.

This now needs a rebase, and some of these issues are no longer relevant. If you are able to rebase and find any additional changes, I will be happy to help review and get the PR merged.

**Comment by camilamacedo86** [comment]
Hi @nemethloci, @stmcginnis

The PR https://github.com/kubernetes-sigs/kubebuilder/pull/3758 was merged. 
So, I think we do not need this one anymore.

However, if you both see that it is still missing points to be addressed, could you please create a new PR to address those or reopen this one if you wish?

Therefore, I hope that you do not mind of we close this one.

**Pull request closed without merging**